setup.cfg

```diff
@@ -65,6 +65,10 @@ skip_glob = tests/testdata
 
 [mypy]
 scripts_are_modules = True
+no_implicit_optional = True
+warn_redundant_casts = True
+warn_unused_ignores = True
```

Comment:
I would remove `warn_unused_ignores = True` just for now. It isn't an issue at the moment. Instead it's more likely to cause false positives (see comment above). Other false-positives are the two `type: ignores` in `rebuilder.py`. They are only necessary in Python 3.9+, thus when checked with 3.8 they are marked as unnecessary.

Reply:
Look like mypy has the same problem than us for useless-suppression. We could mimic how they handle it ?

Reply:
> Look like mypy has the same problem than us for useless-suppression. We could mimic how they handle it ?

We already do ;) They don't handle it.

Reply:
I mean they have an option to activate it or not with ``warn_unused_ignores = True``, like we do when we enable useless-suppression, but do you know how they handle discrepancies that appear between python interpreter when unused ignores are activated ?

Reply:
If the code is evaluated, they emit an error. I.e. not if it's guarded behind a `if sys.version_info` check what is quite common when using typing.